Frontend AOPP UI

thisconnect opened the pull request.

thisconnect commented:
Will open a separate PR for "rewrite select component to typescript" (1217ece)

benma commented on frontends/web/src/routes/account/send/send.tsx:
```diff
@@ -59,7 +58,7 @@ interface SignProgress {
 
 interface State {
-    account?: Account;
+    account?: accountApi.IAccount;
```
i am confused, this change is already in staging-aopp hmm

thisconnect replied:
?

I see `Account` https://github.com/digitalbitbox/bitbox-wallet-app/blob/staging-aopp/frontends/web/src/routes/account/send/send.tsx#L62

benma replied:
Ah right, I was looking at the wrong line, sorry. Where was `Account` defined? I must be blind.

thisconnect replied:
you probably mean this here in the props interface https://github.com/digitalbitbox/bitbox-wallet-app/pull/1373/files#diff-b78d1ac6efeedc528ab327396d435b97de9497cd9354a099a1517cf50fc076ceR47

benma reviewed (commented):
> Will open a separate PR for "rewrite select component to typescript" ([1217ece](https://github.com/digitalbitbox/bitbox-wallet-app/commit/1217ece7ccf1410bb15999807d427c0b7d715616))

Also the commit about icons and SVGs should be in another PR - smaller PRs = better PRs :D

thisconnect commented:
removed "add dialog button" commit and opened a new PR to master https://github.com/digitalbitbox/bitbox-wallet-app/pull/1378

(this was first used for the await user confirmation dialog, but now we changed to fullscreen)

thisconnect commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -34,85 +40,189 @@ interface SubscribedProps {
+                                {/* TODO: show address again on the device */}
+                                <Button secondary onClick={() => console.warn('TODO: show address on device again')}>
+                                    {t('aopp.reverify')}
+                                </Button>
```
@benma for you :)

thisconnect commented:
opened each commit in "smaller PRs = better PRs :D"

- move commonly used icons to icon component https://github.com/digitalbitbox/bitbox-wallet-app/pull/1379
- revamp secondary button style https://github.com/digitalbitbox/bitbox-wallet-app/pull/1381
- rewrite more form components to typescript https://github.com/digitalbitbox/bitbox-wallet-app/pull/1380

this PR requires https://github.com/digitalbitbox/bitbox-wallet-app/pull/1379

but https://github.com/digitalbitbox/bitbox-wallet-app/pull/1382 depends on this PR

thisconnect commented:
Depends on https://github.com/digitalbitbox/bitbox-wallet-app/pull/1380 (rebased this PR on frontend-typescript-forms)

benma commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -34,85 +40,189 @@ interface SubscribedProps {
+    }
+
+    public componentDidUpdate(prevProps) {
+        if (this.props.aopp?.accounts !== prevProps.aopp?.accounts) {
```
I don't think this check works, as two arrays are compared. I think it's only a reference comparison, not a value comparison. What about `const accountsKey = accounts => accounts.map(a => a.code).join()` and then compare `accountsKey(this.props.aopp.accounts) !== accountsKey(prevProps.aopp.accounts)` or similar?

thisconnect replied:
💯 , what do you think about using `utils/equal` https://github.com/digitalbitbox/bitbox-wallet-app/pull/1373/commits/78068e45f1a922c5e4d63abf189b15a9a3371320#diff-c4c5f0fa92cdc4f4a6fbdd048df4be6bea47bea9db5204b4656e7ac59ba96869R57

benma commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -34,85 +40,189 @@ interface SubscribedProps {
+        e.preventDefault();
+    }
+
+    private Banner = ({ children }) => (<div className={styles.banner}>{children}</div>);
+    private Content = ({ children }) => (<div className={styles.content}>{children}</div>);
+    private Header = (props) => (
+        <header className={styles.header}>
+            <h1 className={styles.title}>{props.title}</h1>
+            {props.children}
+        </header>
+    );
+
```
Please move them out of the class, as they don't need any class props/state/vars. Also add types if possible.

thisconnect replied:
Moved Header and Content to the fullscreen module and added types.

Kept Banner in this module, but moved out of the class.

PTAL if ok I'll squash into this commit
https://github.com/digitalbitbox/bitbox-wallet-app/pull/1373/commits/78068e45f1a922c5e4d63abf189b15a9a3371320

benma replied:
:heavy_check_mark:

benma commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -34,85 +40,189 @@ interface SubscribedProps {
+                        </Fullscreen>
+                    );
+                }
+                // show syncing message when no aopp.accounts are found
```
if the state is choosing-accounts, there is always at least one account, otherwise the backend would jump to the error state.

maybe there is a way to encode it in the types, but otherwise feel free to just `!` or `?` it or whatever the syntax is to override typescript with your better knowledge :)

thisconnect replied:
I don't like https://github.com/digitalbitbox/bitbox-wallet-app/pull/1373/commits/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2

will look if there is an easy nicer solution in the interface.

thisconnect replied:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/compare/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2..a11ea8a3885e5e2a801d19718ae46d6bfb0b5d2b

benma reviewed (approved):
utACK, but please check the comments

thisconnect commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -121,44 +121,33 @@ class Aopp extends Component<Props, State> {
-                }
-                // show syncing message when no aopp.accounts are found
-                // technically we don't need this return and could let it fall to case 'syncing'
+                // eslint-disable-next-line no-case-declarations
```
:(

thisconnect commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -121,44 +121,33 @@ class Aopp extends Component<Props, State> {
+                                <Select
+                                    label={t('buy.info.selectLabel')}
+                                    options={options}
+                                    defaultValue={options![0].value}
```
☝️

thisconnect replied:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/compare/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2..a11ea8a3885e5e2a801d19718ae46d6bfb0b5d2b

thisconnect commented on frontends/web/src/components/aopp/aopp.tsx:
```diff
@@ -121,44 +121,33 @@ class Aopp extends Component<Props, State> {
-                // show syncing message when no aopp.accounts are found
-                // technically we don't need this return and could let it fall to case 'syncing'
+                // eslint-disable-next-line no-case-declarations
+                const options = aopp?.accounts?.map(account => {
```
☝️

thisconnect replied:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/compare/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2..a11ea8a3885e5e2a801d19718ae46d6bfb0b5d2b
